### ISSUE TITLE
Make JRuby available on SDKMAN!

### ISSUE THREAD

**Issue by marc0der** [jruby/jruby#3511]
SDKMAN is the SDK Manager, a tool used at large by the alternate JVM language crowd for installing SDKs like Groovy, Scala, Ceylon, Grails, Gradle, SBT and many more. It's similar to rvm, but is focused on all JVM languages.

It would be great to have JRuby on there too. This would involve the JRuby team pushing their own releases to SDKMAN via the Vendor API. It usually involves 3 REST calls hooked into your release build pipeline. Details of the API and process can be found on the [SDKMAN Vendors](http://sdkman.io/vendors.html) page.

**Comment by helio-frota** [comment]
+1

**Comment by swistaczek** [comment]
+1

**Comment by wfouche** [comment]
+1

**Comment by headius** [comment]
Ok, finally catching up with some of my December backlog.

I'd love to look into supporting this, so thanks for the issue and documentation link! I've joined the Discord "vendors" channel to get some help with the onboarding process. Our releases are all pushed to Maven, and all that's necessary to get a runnable JRuby install is to fetch the `jruby-dist` artifact, unpack it, and put its `bin` dir in `path`. There's other forms of deployment depending on what makes the most sense for SDKMAN, but I think the full distribution is the right path.

Help wanted!

**Comment by wfouche** [comment]
JReleaser can publish a release to SDKMAN. This might be the simplest way to publish JRuby to SDKMAN.

- https://jreleaser.org/guide/latest/reference/packagers/sdkman.html

**Comment by headius** [comment]
Perhaps @aalmiray can show me how to integrate it into our release process?

**Comment by aalmiray** [comment]
There's a sdkman-maven-plugin as well (made by yours truly) which could be used with the main repo/distro as it relies on Maven for build. 

Alternatively, JReleaser may be used as well.

**Comment by headius** [comment]
The Maven plug-in approach seems like it would fit best into our workflow. I can have a look at that and see what's required.

**Comment by headius** [comment]
I see there's also a GitHub action that might be even simpler since we already release to Maven Central.

**Comment by aalmiray** [comment]
If you are referring to https://github.com/sdkman/sdkman-action then that only works for consuming candidates, not publishing them.

**Comment by headius** [comment]
I was looking at https://github.com/sdkman/sdkman-release-action.

**Comment by aalmiray** [comment]
Ah, that's even better 👍

**Comment by aalmiray** [comment]
FWIW sdkman requires its candidates to provide Zip archives that follow this layout https://jreleaser.org/guide/latest/concepts/distributions/java-binary.html

**Comment by headius** [comment]
Luckily that's how JRuby's dist is already structured.

* bin/jruby and bin/ruby are a shell script launchers
* bin/jruby.bat is the same for Windows (and there's a bin/jruby.exe as well)
* lib/jruby.jar is the main JRuby jar

Of course there's tons of other stuff, like the entire Ruby standard library and bundled gems, but the base layout seems to be compatible!

**Comment by marc0der** [comment]
Yep, the GitHub release action is a new addition and makes life a lot easier. You would need to use it in conjunction with the [sdkman-default-action](https://github.com/sdkman/sdkman-default-action) to set the released version as the new default. It still needs some docs, sorry about that 😄

**Comment by wfouche** [comment]
Just to get the ball rolling and get an initial version of JRuby uploaded and released via SDKMAN, using `curl` might be the simplest way of getting this done.

https://sdkman.io/vendors/

*Release a new Candidate Version of Groovy (Example)*

```
curl -X POST \
-H "Consumer-Key: CONSUMER_KEY" \
-H "Consumer-Token: CONSUMER_TOKEN" \
-H "Content-Type: application/json" \
-H "Accept: application/json" \
-d '{"candidate": "groovy", "version": "2.4.2", "url": "https://groovy.jfrog.io/artifactory/dist-release-local/groovy-zips/apache-groovy-binary-2.4.6.zip"}' \
https://vendors.sdkman.io/release
```

**Comment by headius** [comment]
Ok I'm looking into this a bit today.
